src/src/com/microsoft/aad/adal/AuthenticationContext.java

```diff
@@ -1915,6 +1932,70 @@ public boolean isConnectionAvailable() {
         }
     }
 
+    protected class BrokerResumeResultReceiver extends BroadcastReceiver
+    {
+        public BrokerResumeResultReceiver() {}
+
+        @Override
+        public void onReceive(Context context, Intent intent) 
+        {
+            final String methodName = ":BrokerInstallLocalReceiver:onReceive";
+            Logger.d(TAG + methodName, "Received result from broker.");
+            final int receivedWaitingRequestId = intent.getIntExtra(AuthenticationConstants.Browser.REQUEST_ID, 0);
+            if (receivedWaitingRequestId == 0)
+            {
+                Logger.v(TAG + methodName, "Received waiting request is 0, error will be thrown, cannot find correct "
+                    + "callback to send back the result.");
+                throw new AuthenticationException(ADALError.AUTH_FAILED_NO_STATE, "Received waiting request id is 0.");
+            }
+          
+            final AuthenticationRequestState waitingRequest = getWaitingRequest(receivedWaitingRequestId);
+            
+            final String errorCode = intent.getStringExtra(AuthenticationConstants.Browser.RESPONSE_ERROR_CODE);
+            if (!StringExtensions.IsNullOrBlank(errorCode))
+            {
+                final String errorMessage = intent.getStringExtra(AuthenticationConstants.Browser.RESPONSE_ERROR_MESSAGE);
+                final String returnedErrorMessage = "ErrorCode: " + errorCode + " ErrorMessage" + errorMessage + getCorrelationInfoFromWaitingRequest(waitingRequest);
+                Logger.v(TAG + methodName, returnedErrorMessage);
+                waitingRequestOnError(waitingRequest, receivedWaitingRequestId, new AuthenticationException(ADALError.AUTH_FAILED, returnedErrorMessage));
+            }
+            else
+            {
+                Logger.v(TAG + methodName, "Received token result from broker reqest resume.");
+                final AuthenticationResult brokerResult = processTokenResultFromBroker(intent);
```

Comment:
Just so we do not forget - how do we ensure that only the intended app can process the token result and not some malicious app.